src/core/ucc_lib.c

```diff
@@ -18,3 +28,70 @@ ucc_status_t ucc_init_version(unsigned api_major_version,
 {
     return UCC_ERR_NOT_IMPLEMENTED;
 }
+
+ucc_status_t ucc_lib_config_read(const char *env_prefix, const char *filename,
+                                 ucc_lib_config_t **config_p)
+{
+    ucc_lib_config_t *config;
+    ucc_status_t      status;
+    int               full_prefix_len;
+    const char       *base_prefix = "UCC_";
+
+    config = ucc_malloc(sizeof(*config), "lib_config");
+    if (config == NULL) {
+        ucc_error("failed to allocated %d bytes for lib_config",
```

Comment:
* allocated -> allocate?
* sizeof returns size_t, so print type should be different